### PR TITLE
Allow non-static references in mocks

### LINKED ISSUE #60: Allow non-static references in mocks

### PR DESCRIPTION
**Pull request by bash**
Resolves #60

### CODE REVIEW

File: mockiato-codegen/src/generate.rs [hunk 1]
```diff
@@ -19,17 +17,41 @@ mod lifetime_rewriter;
...
 
-pub(crate) fn generate_mock(
-    mockable_attr: MockableAttr,
-    item_trait: &ItemTrait,
-    trait_decl: &TraitDecl,
-) -> TokenStream {
-    let mock_struct_ident = mock_struct_ident(&trait_decl, mockable_attr.name_attr);
+#[derive(Debug, Default)]
+pub(crate) struct GenerateMockOptions {
+    pub(crate) mock_struct_ident: Option<Ident>,
```
Comment (jnferner) [inline mockiato-codegen/src/generate.rs]: Naming suggestion: `custom_struct_ident`

File: mockiato-codegen/src/generate/constant.rs [hunk 2]
```diff
@@ -20,17 +19,13 @@ pub(super) fn arguments_lifetime() -> Lifetime {
...
     const IDENTIFIER_POSTFIX: &str = "Mock";
 
-    name_attr.map(|attr| attr.ident).unwrap_or_else(|| {
-        Ident::new(
-            &format!("{}{}", trait_decl.ident, IDENTIFIER_POSTFIX),
-            trait_decl.span.into(),
-        )
-    })
+    Ident::new(
+        &format!("{}{}", trait_decl.ident, IDENTIFIER_POSTFIX),
```
Comment (jnferner) [inline mockiato-codegen/src/generate/constant.rs]: A "postfix" is usually called a "suffix"

File: mockiato-codegen/src/parse/mockable_attr.rs [hunk 3]
```diff
@@ -11,11 +12,15 @@ pub(crate) struct MockableAttr {
     /// The name sub-attribute. Example: `#[name = "FooMock"]`
     /// This customizes the name of the generated mock struct.
     pub(crate) name_attr: Option<NameAttr>,
+    /// The static sub-attribute. Example: `#[mockable(static)]`.
+    /// Enforces that only static lifetimes are used within the mock.
+    pub(crate) static_attr: Option<StaticAttr>,
```
Comment (jnferner) [inline mockiato-codegen/src/parse/mockable_attr.rs]: When writing `#[mockable(static)]`, I intuitively think of static objects. Could we rename this sub-attribute to something like `static_references`?

File: mockiato-codegen/src/generate/constant.rs (hunk 2 above)
Comment (bash) [reply mockiato-codegen/src/generate/constant.rs]: That's not an issue with my new code, but rather an existing issue that affects (most) code in this file.
I created an issue: https://github.com/myelin-ai/mockiato/issues/82

File: mockiato-codegen/src/parse/mockable_attr.rs (hunk 3 above)
Comment (bash) [reply mockiato-codegen/src/parse/mockable_attr.rs]: Addressed in 9709405

File: mockiato-codegen/src/generate.rs (hunk 1 above)
Comment (bash) [reply mockiato-codegen/src/generate.rs]: Addressed in 67d2d7c

**Pull request merged**